### ISSUE TITLE
[HOLD for payment 2024-08-09] [$250]  Mention List - Users are not alphabetically ordered

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#43845]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.84-2
**Reproducible in staging?:** Y
**Reproducible in production?:** N 
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** applausetester+mentiont1@applause.expensifail.com
**Issue reported by:** Applause - Internal Team

**Issue found when executing PR** https://github.com/Expensify/App/pull/42553
## Action Performed:
Precondition: Sign in with a high-trafic account. Create a workspace room and group. Then invite the following users [natnael.expensify+idf6yd@gmail.com,](mailto:natnael.expensify+idf6yd@gmail.com,)
[natnael.expensify+lajf088b@gmail.com,](mailto:natnael.expensify+lajf088b@gmail.com,)
[natnael.expensify+ldjf66vd@gmail.com,](mailto:natnael.expensify+ldjf66vd@gmail.com,)
[natnael.expensify+3@gmail.com](mailto:natnael.expensify+3@gmail.com)
to the group and the room. The accounts are set up for this test.
1. Navigate to the the workspace room chat > type @
2. Navigate to the group chat > type @

## Expected Result:
Users should be listed alphabetically on mention list as thy are on members page

## Actual Result:
User name that starts with upper case comes before user names that stars with lower case even if they shouldn't if ordered alphabetically

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/d7568c2b-f66c-4bee-9c08-55067c4c7850

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01c4e1646d5e188fdf</li>
        <li>Upwork Job ID: 1802726751851213312</li>
        <li>Last Price Increase: 2024-07-08</li>
<li>Automatic offers: </li>
<ul>
<li>situchan | Reviewer | 103067158</li>
<li>dominictb | Contributor | 103067159</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @cead22</details>

**Comment by lanitochka17** [comment]
@cead22  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #vip-vsp

**Comment by situchan** [comment]
reviewing...

**Comment by situchan** [comment]
Thanks for the proposals. Please link offending PR in RCA as this was deploy blocker.

**Comment by situchan** [comment]
@dominictb I think your solution is unnecessarily complex. We should keep new sorting logic introduced in https://github.com/Expensify/App/pull/42553.

**Comment by situchan** [comment]
@devguest07's [proposal](https://github.com/Expensify/App/issues/43845#issuecomment-2173784967) looks good as their solution is enough to fix the root cause.
Please make sure that there's no TS error.
🎀👀🎀 C+ reviewed

**Comment by situchan** [comment]
@dominictb please take any example

**Comment by situchan** [comment]
I think we should first determine if `lodashSortBy` or `localeCompare` should be used for sorting names.

**Comment by situchan** [comment]
> > I think we should first determine if lodashSortBy or localeCompare should be used for sorting names.
> 
> @situchan Sorry do you mean

> > `lodashSortBy` or `localeCompare`

General sorting logic I mean

**Comment by situchan** [comment]
The solution is being discussed

**Comment by situchan** [comment]
ok I am fine with making all name sorting logic consistent with `localeCompare`. (or `lodashSortBy`?)
@cead22 wdyt?

**Comment by cead22** [comment]
Can you summarize what we're deciding? I'm always for consistency, but can you share the details in #expensify-open-source and we can decide there? Include the problem, the root cause, what we do in other places, and your proposal of what we should do here

**Comment by situchan** [comment]
@dominictb do you mind posting in slack?

**Comment by cead22** [comment]
> ok I am fine with making all name sorting logic consistent with `localeCompare`. (or `lodashSortBy`?) @cead22 wdyt?

This works for me, so long as the names show up like the do in the member's page of the video
<img width="528" alt="image" src="https://github.com/Expensify/App/assets/165133/91042607-ca69-4f75-8f2e-ec0ad2a1a5f6">

**Comment by cead22** [comment]
@situchan are there any other blockers on the proposal besides the point above?

**Comment by situchan** [comment]
> There's no other blockers/concerns, @situchan can confirm 👍

👍 

🎀👀🎀 (ref: https://github.com/Expensify/App/issues/43845#issuecomment-2189642700)

**Comment by cead22** [comment]
C+ review is still in progress

**Comment by OfstadC** [comment]
### Payment Summary:

- Paid @situchan $250 via [UpWork](https://www.upwork.com/nx/wm/workroom/37781357/overview?nav_dir=pop)
- Paid @dominictb $250 via [Upwork](https://www.upwork.com/nx/wm/workroom/37820155/overview?nav_dir=pop)
